### PR TITLE
add token request and use npm instead of yarn

### PR DESCRIPTION
**Pull request by KeithWyatt**
- Start using npm instead of yarn
- Added the token endpoint to enable OAuth use
- Start sending Bearer token header with other requests

### CODE REVIEW

**Comment by KeithWyatt** [comment]
Since we accept node 14 & 16 need a code owner to update the required PR build checks to match

**Review by MaxGoo — APPROVED** [review]
Nice, thanks for cleaning up this library as well.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,28 +6,44 @@ jobs:
...
-      - uses: actions/setup-node@v1
         with:
-          node-version: 12
-      - name: Get yarn cache
-        id: yarn-cache
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-      - name: Cache node modules
+          fetch-depth: 0
+      - run: |
+          git fetch --prune
```
Comment (iansu) [inline .github/workflows/ci.yml:17]: Don't need this

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -6,28 +6,44 @@ jobs:
...
+      - name: Cache mongo memory server
         uses: actions/cache@v1
         with:
-          path: ${{ steps.yarn-cache.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          path: ~/.cache/mongodb-binaries
+          key: ${{ runner.os }}-node-mongodb
           restore-keys: |
-            ${{ runner.os }}-yarn-
+            ${{ runner.os }}-node-mongodb
```
Comment (iansu) [inline .github/workflows/ci.yml:28]: Don't need anything related to mongo memory server

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -6,28 +6,44 @@ jobs:
...
-        run: yarn format:check
+        run: npm run format:check
       - name: Lint
-        run: yarn lint
+        run: npm run lint
       - name: Build
-        run: yarn build
+        run: npm run build
+        if: github.ref != 'refs/heads/master'
+      - name: Build all
```
Comment (iansu) [inline .github/workflows/ci.yml:41]: "Build all" and "Test all" aren't needed

**Review by iansu — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -8,22 +8,30 @@ A Flinks API wrapper for Node.js
...
 
-`yarn add node-flinks`
+`npm install node-flinks`
 
 ### Creating a client
 
 ```ts
 import FlinksClient from 'node-flinks';
 
-const flinks = new FlinksClient({
```
[the remark this answers is not in the document]
Comment (KeithWyatt) [reply README.md:18]: never needed them in the first place.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,13 @@
 # Node Flinks Changelog
 
+## 0.4.0 (August 4, 2022)
+
+- Start using `npm` instead of `yarn`
+
+## 0.3.0 (July 28 4, 2022)
+
+- Add `token` command
```
[the remark this answers is not in the document]
Comment (KeithWyatt) [reply CHANGELOG.md:9]: published 3.0 but didn't release it. 
image

**Review by iansu — CHANGES_REQUESTED** [review]

**Review by iansu — APPROVED** [review]

**Pull request merged**